system/lib/pthread/library_pthread.c

```diff
@@ -640,7 +634,7 @@ void emscripten_current_thread_process_queued_calls() {
   pthread_mutex_unlock(&call_queue_lock);
 
   // If the queue was full and we had waiters pending to get to put data to queue, wake them up.
-  emscripten_futex_wake((void*)&q->call_queue_head, 0x7FFFFFFF);
+  emscripten_futex_wake((void*)&q->call_queue_head, INT_MAX);
```

Comment:
This can be part of the cleanup PR too.

Reply:
Reverted with commit 165078536c1470685ac7a76db7868d5d601cdbf0, I'll make a PR with these changes only.